domain/resource/service/resource.go

```diff
@@ -695,8 +695,8 @@ func (s *Service) DeleteResourcesAddedBeforeApplication(ctx context.Context, res
 //     be found in the database.
 //   - [resourceerrors.OriginNotValid] if the resource origin is not valid.
 func (s *Service) ImportResources(ctx context.Context, args resource.ImportResourcesArgs) error {
-	resourceNames := make(map[string]bool)
 	for _, appArg := range args {
+		resourceNames := make(map[string]bool)
```

Comment:
todo: add a test for this.

Reply:
Already got one, added another to check that two applications with resources of the same name pass fine.